Client/mods/deathmatch/logic/lua/CLuaFunctionDefs.World.cpp

```diff
@@ -1920,3 +1922,139 @@ int CLuaFunctionDefs::FetchRemote(lua_State* luaVM)
     lua_pushboolean(luaVM, false);
     return 1;
 }
+
+// table getRemoteRequests([resource theResource = nil])
+int CLuaFunctionDefs::GetRemoteRequests(lua_State* luaVM)
+{
+    CScriptArgReader argStream(luaVM);
+    CResource*       pResource = nullptr;
+    CLuaMain*        pLuaMain = nullptr;
+    int              iIndex = 0;
+
+    argStream.ReadUserData(pResource, NULL);
+
+    // Grab virtual machine
+    if (pResource)
+        pLuaMain = pResource->GetVM();
+
+    lua_newtable(luaVM);
+    for (const auto& request : g_pClientGame->GetRemoteCalls()->GetCalls())
+    {
+        if(!pLuaMain || request->GetVM() == pLuaMain)
+        {
+            lua_pushnumber(luaVM, ++iIndex);
+            lua_pushuserdata(luaVM, request);
+            lua_settable(luaVM, -3);
+        }
+    }
+
+    return 1;
+}
+
+// table getRemoteRequestInfo(element theRequest)
+int CLuaFunctionDefs::GetRemoteRequestInfo(lua_State* luaVM)
+{
+    CScriptArgReader argStream(luaVM);
+    CLuaArguments    info, requestedHeaders;
+    CRemoteCall*     pRemoteCall = nullptr;
+    CResource*       pThisResource = g_pClientGame->GetResourceManager()->GetResourceFromLuaState(luaVM);
+
+
+    argStream.ReadUserData(pRemoteCall);
+
+    if (!argStream.HasErrors())
+    {
+        CResource* pResource = pRemoteCall->GetVM()->GetResource();
+        bool bExtendedInfo = (pResource == pThisResource);
+        
+        info.PushString("type");
+        info.PushString((pRemoteCall->IsFetch() ? "fetch" : "call"));
+
+        // TODO: remove query_string from url when bExtendedInfo isn't set
+        info.PushString("url");
+        info.PushString(pRemoteCall->GetURL().c_str());
+
+        info.PushString("queue");
+        info.PushString(pRemoteCall->GetQueueName().c_str());
+        
+        info.PushString("resource");
+        
+        if (pResource)
+            info.PushResource(pResource);
+        else
+            info.PushBoolean(false);
+
+        info.PushString("start");
+        info.PushNumber(pRemoteCall->GetStartTime());
```

Comment:
I think we should return the number of ticks since the download was queued

Reply:
Good idea, gonna change that later today.